test/ha/ha.go

```diff
@@ -44,47 +41,38 @@ const (
 	haReplicas       = 2
 )
 
-func getLeader(t *testing.T, clients *test.Clients, labelSelector string) (string, error) {
-	var leader string
-	if err := wait.PollImmediate(test.PollInterval, time.Minute, func() (bool, error) {
-		controllerPodList, err := clients.KubeClient.Kube.CoreV1().Pods(servingNamespace).List(metav1.ListOptions{
-			LabelSelector: labelSelector,
-		})
-		if err != nil {
-			return false, fmt.Errorf("error retrieving pods with label %s: %w", labelSelector, err)
-		}
-		for _, pod := range controllerPodList.Items {
-			podLogs, err := logsAsString(clients.KubeClient.Kube.CoreV1().Pods(servingNamespace).GetLogs(pod.Name, &corev1.PodLogOptions{}))
-			if err != nil {
-				if strings.Contains(err.Error(), "ContainerCreating") {
-					// Go ahead and check other pods
-					continue
+func getLeader(t *testing.T, clients *test.Clients, component, labelSelector string) (string, error) {
+	watcher, err := clients.KubeClient.Kube.CoreV1().Events(servingNamespace).Watch(metav1.ListOptions{
```

Comment:
Why not just poll/watch `v1.Lease` objects and check `holderIdentity` field?

Reply:
Actually, polling the Lease is simpler and cleaner. Thanks, lesson learned. Add one more commit with the changes.